src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -83,7 +92,7 @@ protected async Task<DValue<RecordValue>> UpdateAllowedFieldsAsync(RecordValue c
                 return await base.UpdateFieldsAsync(changeRecord, cancellationToken).ConfigureAwait(false);
             }
 
-            await foreach (var field in changeRecord.GetFieldsAsync(cancellationToken).ConfigureAwait(false))
+            await foreach (var field in changeRecord.GetOriginalFieldsAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
>GetOriginalFieldsAsync [](http://example.com/codeflow?start=53&length=22)

Can you explain this change? Why didn't the previous code work?

Reply:
`GetFieldsAsync` will return all fields based on the record type. For instance, the expression `Table({a:1},{b:"1"})` will return `*[a:w, b:s]`.
Now, if we traverse the records and iterate using `GetFieldsAsync`, we will get both `a` and `b` fields, which will return `Blank()` depending on the record. This could potentially update unwanted fields in the datasource to `Blank()`.

Example (bug):
`Set(t1, Table({a:1,b:1,c:1},{a:2,b:2,c:2}))`
`Patch(t1, Table({a:1, b:99}))`
`t1 => Table({a:1,b:99,c:Blank()},{a:2,b:2,c:2})`